x/persistent_test.go

```diff
@@ -1,36 +1,54 @@
 package x
 
 import (
+	"reflect"
 	"testing"
 
 	"github.com/iov-one/weave/coin"
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
+	"github.com/iov-one/weave/weavetest/assert"
 )
 
 func TestPersistent(t *testing.T) {
-	c := &coin.Coin{Whole: 52, Fractional: 12345, Ticker: "FOO"}
-	bad := &coin.Coin{Whole: 52, Fractional: -12345, Ticker: "of"}
-	should, err := c.Marshal()
-	require.NoError(t, err)
-
-	// marshal
-	bz := MustMarshal(c)
-	assert.Equal(t, should, bz)
+	good := coin.NewCoinp(52, 12345, "FOO")
+	rawGood, err := good.Marshal()
+	assert.Nil(t, err)
+	assert.Equal(t, rawGood, MustMarshal(good))
+
+	bad := coin.NewCoinp(52, -12345, "of")
 	garbage := MustMarshal(bad)
-	assert.NotEqual(t, should, garbage)
+	if reflect.DeepEqual(rawGood, garbage) {
+		t.Fatal("garbage serialization worked")
+	}
 	copy(garbage, []byte{17, 34, 56})
 
-	// unmarshal
-	got := new(coin.Coin)
-	MustUnmarshal(got, bz)
-	assert.Equal(t, c, got)
-	assert.Panics(t, func() { MustUnmarshal(got, garbage) })
-
-	// validate
-	assert.Panics(t, func() { MustValidate(bad) })
-	assert.NotPanics(t, func() { MustValidate(c) })
-	assert.Panics(t, func() { MustMarshalValid(bad) })
-	rebz := MustMarshalValid(c)
-	assert.Equal(t, should, rebz)
+	var got coin.Coin
+	MustUnmarshal(&got, rawGood)
+	assert.Equal(t, good, &got)
+
+	assertPanics(t, func() {
+		MustUnmarshal(&got, garbage)
+	})
+
+	assertPanics(t, func() {
+		MustValidate(bad)
+	})
+
+	MustValidate(good)
+
+	assertPanics(t, func() {
+		MustMarshalValid(bad)
+	})
+
+	rebz := MustMarshalValid(good)
+	assert.Equal(t, rawGood, rebz)
+}
+
+func assertPanics(t testing.TB, fn func()) {
```

Comment:
Didn't you just put this into weavetest? (Oh, I guess other PR).
Please rebase on master once the other PR is merged, and remove this duplicate code before merging this PR